rust/core-lib/src/editor.rs

```diff
@@ -361,6 +361,11 @@ impl Editor {
         self.increment_revs_in_flight();
 
         {
+            let new_len = delta.new_document_len();
+            let delta = match delta.inserts_len() > MAX_SIZE_LIMIT {
```

Comment:
This logic is not a hard bound on the size of the encoded delta, as 1M deletes would come back as 0. Easiest thing to do would be to add 10 * els.len(). I'd also change the name to something like len_estimate, with a comment that it's an estimate of the encoded size of the delta.

Reply:
I'm a bit confused. My thinking here was that we're only concerned with the byte-size of the RPC, and in almost all cases RPC length is determined by the contents of an insert.

I _can_ imagine a scenario where in a very long document, an editor could diligently create a delta that deleted every second character, producing a delta with ~len(docu)/2 delete ops; but it isn't clear to me that we are explicitly trying to defend against a DOS; in any case I think I'm probably misunderstanding your point? 😁

If we're interested, I could definitely write a heuristic that takes into account (say) the number of delete ops, and approximates the actual size of the encoded delta...